source/css/sass/_mixins--spacing.scss

```diff
@@ -1,44 +1,156 @@
+@import "variables";
+
+@mixin _padding-left($size-in-px) {
+  @if $size-in-px == 0 {
+    padding-left: 0;
+  } @else {
+    padding-left: #{$size_in_px}px;
+    padding-left: #{get-rem-from-px($size_in_px)}rem;
+  }
+}
+
+@mixin _padding-right($size-in-px) {
+  @if $size-in-px == 0 {
+    padding-right: 0;
+  } @else {
+    padding-right: #{$size_in_px}px;
+    padding-right: #{get-rem-from-px($size_in_px)}rem;
+  }
+}
 
 @mixin padding($size_in_px, $dimension: "") {
-  @if $dimension == top or $dimension == right or $dimension == bottom or $dimension == left {
+  @if $dimension == "" {
     @if $size_in_px == 0 {
-      padding-#{$dimension}: 0;
+      padding: 0;
     } @else {
-      padding-#{$dimension}: #{$size_in_px}px;
-      padding-#{$dimension}: #{get-rem-from-px($size_in_px)}rem;
+      padding: #{$size_in_px}px;
+      padding: #{get-rem-from-px($size_in_px)}rem;
     }
   }
-  @else if $dimension == "" and type_of($size_in_px) == list {
-    $parts-px: ();
-    $parts-rem: ();
-    @each $size in $size_in_px {
-      @if $size == 0 {
-        $parts-px: append($parts-px, 0, "space");
+  @if $dimension == "inline" {
+    @if $size_in_px == 0 {
+      padding-left: 0;
+      padding-right: 0;
+      padding-inline: 0;
+    } @else if type_of($size_in_px) == list and length($size_in_px) >= 2 {
+
+      $firstValue: nth($size_in_px, 1);
+      $secondValue: nth($size_in_px, 2);
+
+      @if $firstValue == $secondValue {
+
+        @include _padding-left($firstValue);
+        @include _padding-right($secondValue);
+
       } @else {
-        $parts-px: append($parts-px, #{$size}px, "space");
+
+        &[dir="ltr"] {
+          @include _padding-left($firstValue);
+          @include _padding-right($secondValue);
+        }
+
+        &[dir="rtl"] {
+          @include _padding-right($firstValue);
+          @include _padding-left($secondValue);
+        }
+
       }
-    }
-    @each $size in $size_in_px {
-      @if $size == 0 {
-        $parts-rem: append($parts-rem, 0, "space");
+
+      $derivedFirstValue: 0;
+      $derivedSecondValue: 0;
+      @if $firstValue != 0 {
+        $derivedFirstValue: #{get-rem-from-px($firstValue)}rem;
+      }
+      @if $secondValue != 0 {
+        $derivedSecondValue: #{get-rem-from-px($secondValue)}rem;
+      }
+      @if $firstValue == $secondValue {
+        padding-inline: $derivedFirstValue;
       } @else {
-        $parts-rem: append($parts-rem, #{get-rem-from-px($size)}rem, "space");
+        &#{$libero-selector-to-force-sass-output-order} {
+          padding-inline: $derivedFirstValue $derivedSecondValue;
+        }
       }
+
+    } @else {
+      @include _padding-left($size_in_px);
+      @include _padding-right($size_in_px);
+      padding-inline: #{get-rem-from-px($size_in_px)}rem;
     }
-    padding: $parts-px;
-    padding: $parts-rem;
   }
-  @else {
+
+  @if $dimension == "inline-start" {
     @if $size_in_px == 0 {
-      padding: 0;
+      &[dir="ltr"] {
```

Comment:
These assume that we have `dir` attributes _everywhere_, which I'm sure will become a pain. (My thinking for Libero code is to make it happen automatically, but probably difficult for anyone else. Or will just be ignored.)

Trying to imagine a proper fallback for `:dir()`. Did a quick search and couldn't find one. Does this work?

```sass
&[dir="ltr"],
[dir="ltr"] :not([dir="rtl"]) &,
[dir="ltr"] > &:not([dir]) {
```

Reply:
> This selector only applies to one element; you cannot use it to exclude all ancestors.

Dang.

Reply:
As `<html>` will always have a `dir` attribute, will look at how to incorporate that into selectors to help with fallbacks.

Reply:
@thewilkybarkid Having dug into this a bit more, I'm not sure a direct fallback for `:dir()` is going to work (although would be delighted to be proved wrong). The problem, as you allude to above, is that how the `:not()` selector works, or rather how it doesn't work. I can't find a way of constructing CSS selectors (other than ':dir()`) to reliably work out what the current reading direction is when it's inferred from an ancestor element's `dir` attribute.

Reply:
I've got this so far: 
https://codepen.io/davidcmoulton/pen/JamRej (comment/uncomment the appropriate HTML element to select the document reading direction).

With `<html dir="ltr">: The div with the text "rtl directly" should only have right padding, but it has left and right padding. The same applies to its child element with the text "nested implicit rtl".
The "nested rtl directly" element should only have right padding but has both right and left padding

Reply:
How about https://codepen.io/thewilkybarkid/pen/MqPbRa?

Reply:
The "nested implicit rtl" element has both left and right padding, it should have only right padding.

Reply:
Ah. Updated to use margins to be clearer. That's the ancestor problem...

Reply:
I think the compromise might be to require a `dir` at the every level _after_ the switch... Shouldn't affect most usage that way.

Might possibility make sense to put multi-direction capability behind a config option? eLife wouldn't need it, for example, so would reduce the CSS.

Reply:
> I think the compromise might be to require a `dir` at the every level _after_ the switch... Shouldn't affect most usage that way.

I think you've found the sweet spot for compromise @thewilkybarkid. The fallback selectors would be manageable, and the attribute load is much less than it'd otherwise be.

Not yet sure about making multi-direction configurable, now the fallback selectors can be simplified, as we still want almost all of the fallback for direction-sensitive spacing using logical properties anyway.